### DISCUSSION TITLE
Windows development setup

### DISCUSSION THREAD

**Discussion by milansoliya4210** [timoore/vsgCs#7] [General]
Can you share detail setup required for Windows based platform ?

**Comment by timoore** [comment]
> Can you share detail setup required for Windows based platform ?

What difficulties are you having? This is what I do:
* vcpkg for installing curl and zlib
* git for Windows for cloning vsgCs, though it's possible that Visual Studio's git would work fine.
* cmake. I add the cmake variable CMAKE_TOOLCHAIN_FILE=C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake ; obviously you would point to your own vcpkg installation. Alternatively you could follow the vcpkg instructions for integrating vckpkg with Visual Studio.
* The latest release of vsgCs will download cesium-native, VSG, and vsgImGui and build them.
* I compile using Microsoft Visual Studio 2022 Community Edition.
* I run vsgCs either from the MSVC debugger or the command prompt.

**Comment by milansoliya4210** [comment]
/ This is warning while fetching git visual studio
![Screenshot 2023-06-28 113855](https://github.com/timoore/vsgCs/assets/40118048/fd65dec3-cd0b-4162-8a94-b33f693ac3a2)

**Reply by timoore** [reply]
Ah, that's a problem with how Tracy is defined as a git submodule. A workaround would be to create an account at github and load up an SSH key so that the git@github.com syntax works, but I'll see about changing the submodule definition to use https.

**Comment by milansoliya4210** [comment]
I have already curl and give me error like 
![Screenshot 2023-06-28 115455](https://github.com/timoore/vsgCs/assets/40118048/c59009a4-cb72-4c63-84fe-cba054f2e061)

**Comment by timoore** [comment]
> I have already curl and give me error like ![Screenshot 2023-06-28 115455](https://user-images.githubusercontent.com/40118048/249373928-c59009a4-cb72-4c63-84fe-cba054f2e061.png)

How have you installed curl? I don't use Visual Studio's builtin CMake, so I really don't know where it is looking for headers and libraries.

**Comment by milansoliya4210** [comment]
Download binaries from https://curl.se/windows/ and give system environment variables to the path, For windows how you are doing? can you share some common steps for windows if possible ?

**Comment by timoore** [comment]
> Download binaries from https://curl.se/windows/ and give system environment variables to the path, For windows how you are doing? can you share some common steps for windows if possible ?

That's not enough; you need to install header files too. I already described my specific setup; I suggest you read the documentation for vcpkg on how to integrate it with Visual Studio.
